### DISCUSSION TITLE
STORE ACCESS-TOKEN BY USER ON DB

### DISCUSSION THREAD

**Discussion by FedericoGaglio** [keycloak/keycloak#19064] [Q&A]
Hi everyone! It would be of interest for me to solve a problem: when I access keycloak from the interface, an access-token is issued to the user (on /token). The question I ask myself is: is there a way to be able to go and persist this data on the db (I use a mysql)? because going to see the various tables that are automatically saved on the db by keycloak, I don't see this access-token saved in any!! thank you all in advance everyone.

**Comment by darius-m** [comment]
[This statement was not factually correct, so it was removed to not confuse more users]

**Reply by FedericoGaglio** [reply]
thank you for the quick reply. Yes, I'm trying to configure the db directly from the cache configuration, but at the moment I'm getting the following error:

Caused by: java.util.concurrent.ExecutionException: org.infinispan.commons.CacheConfigurationException: ISPN000327:

**Comment by sschu** [comment]
Why exactly would you want Keycloak to store the Tokens it emits?

**Reply by FedericoGaglio** [reply]
it's needed in my organization for a security reasons.

**Reply by sschu** [reply]
Depending on configuration, Keycloak creates a lot of potentially short-lived tokens and storing them all might create a huge pile of data. Not sure what security requirement exactly you can fulfil by storing them. Also, AFAIK they are also not stored currently, not even in memory. At least I don't see a `tokens` cache. There is an `actionTokens` cache but that one stores something different.

**Reply by darius-m** [reply]
I was indeed thinking of `actionTokens`, but if that data is not relevant, I will retract my statement. Thank you for correcting. I will remove that comment.
